sunkit_image/coalignment/tests/test_coalignment.py

```diff
@@ -0,0 +1,110 @@
+
+import sunpy.map
+from sunpy.net import Fido, attrs as a
+from sunpy.util.exceptions import SunpyUserWarning
+
+from sunkit_image.coalignment import coalign
+from sunkit_image.coalignment.decorators import register_coalignment_method, registered_methods
+from sunkit_image.coalignment.interface import AffineParams
+from sunkit_image.data.test import get_test_filepath
+from sunkit_image.tests.helpers import figure_test
+
+@pytest.fixture()
+def eis_test_map():
+    return sunpy.map.Map(get_test_filepath("eis_20140108_095727.fe_12_195_119.2c-0.int.fits"))
+
+
+@pytest.mark.remote_data()
+@pytest.mark.filterwarnings("ignore:transforming other coordinates:astropy.coordinates.errors.NonRotationTransformationWarning")
+def test_coalignment(eis_test_map):
+    aia193_test_map = sunpy.map.Map(Fido.fetch(Fido.search(a.Time(start=eis_test_map.meta["date_beg"], near=eis_test_map.meta["date_avg"], end=eis_test_map.meta["date_end"]), a.Instrument('aia'), a.Wavelength(193*u.angstrom))))
+    # Synchronize rsun to reduce transformation issues
+    aia193_test_map.meta['rsun_obs'] = eis_test_map.meta['dsun_obs']/2
+    nx = (aia193_test_map.scale.axis1 * aia193_test_map.dimensions.x) / eis_test_map.scale[0]
+    ny = (aia193_test_map.scale.axis2 * aia193_test_map.dimensions.y) / eis_test_map.scale[1]
+    aia193_test_downsampled_map = aia193_test_map.resample(u.Quantity([nx, ny]))
+    coaligned_is_map = coalign(aia193_test_downsampled_map, eis_test_map, "match_template")
+    assert coaligned_is_map.data.shape == eis_test_map.data.shape
+    assert_allclose(coaligned_is_map.wcs.wcs.crval[0], aia193_test_downsampled_map.wcs.wcs.crval[0],rtol = 1e-2, atol = 0.13)
+    assert_allclose(coaligned_is_map.wcs.wcs.crval[1], aia193_test_downsampled_map.wcs.wcs.crval[1],rtol = 1e-2, atol = 0.13)
+
+
+@pytest.fixture()
+def cutout_map(aia171_test_map):
+    aia_map = sunpy.map.Map(aia171_test_map)
+    bottom_left = SkyCoord(-300 * u.arcsec, -300 * u.arcsec, frame = aia_map.coordinate_frame)
+    top_right = SkyCoord(800 * u.arcsec, 600 * u.arcsec, frame = aia_map.coordinate_frame)
+    cutout_map = aia_map.submap(bottom_left, top_right=top_right)
+    return cutout_map
+
+
+def test_coalignment_reflects_pixel_shifts(cutout_map, aia171_test_map):
+    """Check if coalignment adjusts world coordinates as expected based on reference coordinate shifts."""
+    messed_map = cutout_map.shift_reference_coord(25 * u.arcsec, 50 * u.arcsec)
+    original_world_coords = cutout_map.reference_coordinate
+    # shifted_world_coords = messed_map.reference_coordinate
+    fixed_cutout_map = coalign(aia171_test_map, messed_map)
+    fixed_world_coords = fixed_cutout_map.reference_coordinate
+    # The actual shifts applied by coalignment should be equal to the expected shifts
+    assert_allclose(original_world_coords.Tx, fixed_world_coords.Tx, rtol=1e-2, atol=0.4)
+    assert_allclose(original_world_coords.Ty, fixed_world_coords.Ty, rtol=1e-2, atol=0.4)
```

Comment:
I think a better test would be whether the difference in the reference coordinates of the fixed and "messed" maps is approximately equal to the amount that the map was shifted by.